miden-lib/asm/shared/util/account_id.masm

```diff
@@ -0,0 +1,387 @@
+# applied, this value defines the public storage mode.
+const.ACCOUNT_ID_STORAGE_MODE_PUBLIC_U32=0 # 0b0000_0000
+
+# Given the least significant 32 bits of an account ID's first felt with the storage mode mask
+# applied, this value defines the private storage mode.
+const.ACCOUNT_ID_STORAGE_MODE_PRIVATE_U32=0x80 # 0b1000_0000
+
+# CONSTANT ACCESSORS
+# =================================================================================================
+
+#! Returns the account storage slot at which faucet data is stored.
+#! Fungible faucet: The faucet data consists of [0, 0, 0, total_issuance]
+#! Non-fungible faucet: The faucet data consists of SMT root containing minted non-fungible assets.
+#!
+#! Inputs:  []
+#! Outputs: [faucet_storage_data_slot]
+#!
+#! Where:
+#! - faucet_storage_data_slot is the account storage slot at which faucet data is stored.
+export.get_faucet_storage_data_slot
+    push.FAUCET_STORAGE_DATA_SLOT
+end
+
+#! Returns the maximum number of account storage slots.
+#!
+#! Inputs:  []
+#! Outputs: [max_num_storage_slots]
+#!
+#! Where:
+#! - max_num_storage_slots is the maximum number of account storage slots.
+export.get_max_num_storage_slots
+    push.MAX_NUM_STORAGE_SLOTS
+end
+
+#! Returns the maximum number of account interface procedures.
+#!
+#! Inputs:  []
+#! Outputs: [max_num_procedures]
+#!
+#! Where:
+#! - max_num_procedures is the maximum number of account interface procedures.
+export.get_max_num_procedures
+    push.MAX_NUM_PROCEDURES
+end
+
+# PROCEDURES
+# =================================================================================================
+
+#! Returns a boolean indicating whether the account is a fungible faucet.
+#!
+#! Inputs:  [acct_id_prefix]
+#! Outputs: [is_fungible_faucet]
+#!
+#! Where:
+#! - acct_id_prefix is the prefix of the account ID.
+#! - is_fungible_faucet is a boolean indicating whether the account is a fungible faucet.
+export.is_fungible_faucet
+    exec.type push.FUNGIBLE_FAUCET_ACCOUNT eq
+    # => [is_fungible_faucet]
+end
+
+#! Returns a boolean indicating whether the account is a non-fungible faucet.
+#!
+#! Inputs:  [acct_id_prefix]
+#! Outputs: [is_non_fungible_faucet]
+#!
+#! Where:
+#! - acct_id_prefix is the prefix of the account ID.
+#! - is_non_fungible_faucet is a boolean indicating whether the account is a non-fungible faucet.
+export.is_non_fungible_faucet
+    exec.type push.NON_FUNGIBLE_FAUCET_ACCOUNT eq
+    # => [is_non_fungible_faucet]
+end
+
+#! Returns a boolean indicating whether the given account_ids are equal.
+#!
+#! Inputs:  [acct_id_prefix, acct_id_suffix, other_acct_id_prefix, other_acct_id_suffix]
+#! Outputs: [is_id_equal]
+#!
+#! Where:
+#! - acct_id_{prefix,suffix} are the prefix and suffix felts of an account ID.
+#! - other_acct_id_{prefix,suffix} are the prefix and suffix felts of the other account ID to compare against.
+#! - is_id_equal is a boolean indicating whether the account IDs are equal.
+export.is_id_eq
```

Comment:
nit: for similar reasons as in the last comment, I wonder if we should rename this to something like `is_eq` (or maybe `are_equal`).

Reply:
Went with `is_equal` in `account_id` and reexporting it as `is_id_equal` from `miden::account`, e.g.
```
export.::miden::util::account_id::is_equal->is_id_equal
```